contractcourt/anchor_resolver.go

```diff
@@ -0,0 +1,170 @@
+	anchor wire.OutPoint
+
+	// resolved reflects if the contract has been fully resolved or not.
+	resolved bool
+
+	// broadcastHeight is the height that the original contract was
+	// broadcast to the main-chain at. We'll use this value to bound any
+	// historical queries to the chain for spends/confirmations.
+	broadcastHeight uint32
+
+	// chanPoint is the channel point of the original contract.
+	chanPoint wire.OutPoint
+
+	// currentReport stores the current state of the resolver for reporting
+	// over the rpc interface.
+	currentReport ContractReport
+
+	// reportLock prevents concurrent access to the resolver report.
+	reportLock sync.Mutex
+
+	contractResolverKit
+}
+
+// newAnchorResolver instantiates a new anchor resolver.
+func newAnchorResolver(anchorSignDescriptor input.SignDescriptor,
+	anchor wire.OutPoint, broadcastHeight uint32,
+	chanPoint wire.OutPoint, resCfg ResolverConfig) *anchorResolver {
+
+	amt := btcutil.Amount(anchorSignDescriptor.Output.Value)
+
+	report := ContractReport{
+		Outpoint:         anchor,
+		Type:             ReportOutputAnchor,
+		Amount:           amt,
+		LimboBalance:     amt,
+		RecoveredBalance: 0,
+	}
+
+	r := &anchorResolver{
+		contractResolverKit:  *newContractResolverKit(resCfg),
+		anchorSignDescriptor: anchorSignDescriptor,
+		anchor:               anchor,
+		broadcastHeight:      broadcastHeight,
+		chanPoint:            chanPoint,
+		currentReport:        report,
+	}
+
+	r.initLogger(r)
+
+	return r
+}
+
+// ResolverKey returns an identifier which should be globally unique for this
+// particular resolver within the chain the original contract resides within.
+func (c *anchorResolver) ResolverKey() []byte {
+	// The anchor resolver is stateless and doesn't need a database key.
+
+	return nil
+}
+
+// Resolve offers the anchor output to the sweeper and waits for it to be swept.
+//
+// NOTE: This function MUST be run as a goroutine.
+func (c *anchorResolver) Resolve() (ContractResolver, error) {
+	anchorInput := input.MakeBaseInput(
+		&c.anchor,
+		input.CommitmentAnchor,
+		&c.anchorSignDescriptor,
+		c.broadcastHeight,
+	)
+
+	// Offer the anchor output to the sweeper. If it is already known by the
+	// sweeper, we will just get another result channel and the previously
+	// set sweeping parameters (fee and exclusive group) will not be
+	// modified. Otherwise we try to sweep with the minimum relay fee. An
+	// exclusive group is not necessary in that case, because we only offer
+	// a single anchor to the sweeper.
+	relayFeeRate := c.ResolverConfig.FeeEstimator.RelayFeePerKW()
+
+	// TODO(joostjager): Probably need to clear the force flag here. Is
+	// there a reason to sweep at a loss at this point?
+	resultChan, err := c.Sweeper.SweepInput(
+		&anchorInput,
+		sweep.Params{
+			Fee: sweep.FeePreference{
+				FeeRate: relayFeeRate,
+			},
+			Force: true,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	select {
+	case sweepRes := <-resultChan:
```

Comment:
Comment from @Roasbeef: In a high fee situation, the sweep may never succeed. Do channels remain pending forever?

Reply:
Addressed via `sweep.ErrTooManyAttempts`